test/AutoDiff/Parse/transpose_attr_parse.swift

```diff
@@ -0,0 +1,92 @@
+// RUN: %target-swift-frontend -parse -verify %s
+
+/// Good
+
+@transpose(of: foo)
+func transpose(v: Float) -> Float
+
+@transpose(of: foo(_:_:))
+func transpose(v: Float) -> Float
+
+@transpose(of: wrt, wrt: 0)
+func transpose(v: Float) -> Float
+
+@transpose(of: foo, wrt: 0)
+func transpose(v: Float) -> Float
+
+@transpose(of: foo, wrt: (0, 1))
+func transpose(v: Float) -> (Float, Float)
+
+@transpose(of: foo, wrt: (self, 0, 1, 2))
+func transpose(v: Float) -> (Float, Float, Float, Float)
+
+// Qualified declaration.
+@transpose(of: A.B.C.foo(x:y:_:z:))
+func transpose(v: Float) -> Float
+
+// Qualified operator.
+// TODO(TF-1065): Consider disallowing qualified operators.
```

Comment:
FWIW, I think that your implemented behavior is correct and nothing needs to change here.  "TF-1065" is really about the existing swift compiler being buggy and not allowing qualified names for operators like ..< . I'd suggest splitting that off to a swift bug, and I'd recommend that operator having to be spelled as Swift.`..<`

Reply:
Thanks for the thorough review! I started a [discussion](https://forums.swift.org/t/operator-member-syntax/31773) regarding operator member syntax with proposed backtick syntax (`` SIMD2<Float>.`.<` ``).

As stated in TF-1065, qualified operator syntax isn't strictly necessary for `@transpose` or `@derivative` attribute because type-checking for these attributes require the operator to be defined in the same type context, where lookup is not ambiguous.

Taking action on these parser changes seems low priority so I'll leave the code and tests as is for now.